src/move.c

```diff
@@ -893,7 +895,7 @@ curs_rows(win_T *wp)
 	}
 	if (valid
 #ifdef FEAT_DIFF
-		&& (lnum != wp->w_topline || !wp->w_p_diff)
+		&& (lnum != wp->w_topline || wp->w_skipcol == 0)
```

Comment:
I think we need to execute the `else` branch when skipcol is non-zero, not just when 'diff' is set. This affects two screendumps which currently contained invalid cursor positions, which I updated. This in combination with the added redraw_later() also fix #12411.

Reply:
This `#ifdef FEAT_DIFF` should be removed then.

Reply:
And the `else` branch is executed when 'diff' is not set.

Reply:
Oops, I read it wrong.

Reply:
Hmm no I think we need to check for skipcol besides checking for 'diff' filler lines, not instead of. Updated.